pulsar-broker/src/main/java/org/apache/pulsar/broker/offload/impl/ManagedLedgerOffloader.java

```diff
@@ -244,35 +315,34 @@ static String indexBlockOffloadKey(long ledgerId, UUID uuid) {
         String indexKey = indexBlockOffloadKey(ledgerId, uid);
         scheduler.chooseThread(ledgerId).submit(() -> {
                 try {
-                    promise.complete(S3BackedReadHandleImpl.open(scheduler.chooseThread(ledgerId),
-                                                                 s3client,
+                    promise.complete(BackedReadHandleImpl.open(scheduler.chooseThread(ledgerId),
+                                                                 blobStore,
                                                                  bucket, key, indexKey,
                                                                  VERSION_CHECK,
                                                                  ledgerId, readBufferSize));
                 } catch (Throwable t) {
+                    log.error("Failed readOffloaded: ", t);
                     promise.completeExceptionally(t);
                 }
             });
         return promise;
     }
 
-    private static void addVersionInfo(ObjectMetadata metadata) {
-        metadata.getUserMetadata().put(METADATA_FORMAT_VERSION_KEY, CURRENT_VERSION);
-        metadata.getUserMetadata().put(METADATA_SOFTWARE_VERSION_KEY,
-                                       PulsarBrokerVersionStringUtils.getNormalizedVersionString());
-        metadata.getUserMetadata().put(METADATA_SOFTWARE_GITSHA_KEY, PulsarBrokerVersionStringUtils.getGitSha());
-    }
+
 
     @Override
     public CompletableFuture<Void> deleteOffloaded(long ledgerId, UUID uid) {
         CompletableFuture<Void> promise = new CompletableFuture<>();
         scheduler.chooseThread(ledgerId).submit(() -> {
             try {
-                s3client.deleteObjects(new DeleteObjectsRequest(bucket)
-                    .withKeys(dataBlockOffloadKey(ledgerId, uid), indexBlockOffloadKey(ledgerId, uid)));
+                blobStore.removeBlob(bucket, dataBlockOffloadKey(ledgerId, uid));
+                blobStore.removeBlob(bucket, indexBlockOffloadKey(ledgerId, uid));
+                // adobe/s3mock not support removeBlobs well.
+                /*blobStore.removeBlobs(bucket,
```

Comment:
Since we are shipping out own s3mock, we should fix it.
And I don't understand how it doesn't support removeBlobs if deleteObjects worked fine.